lib/puppet/pops/types/execution_result.rb

```diff
@@ -104,6 +104,10 @@ def _pcore_init_hash
       @result_hash
     end
 
+    def ==(another_result)
+      self.result_hash == another_result.result_hash
```

Comment:
Not sure if there's a way I should be testing that another_result is also an ExecutionResult. I guess compare `.class`?.

Reply:
Updated to do that.